File: ShortenLink/src/main/java/com/ieng/shortenLink/entity/ShorterLink.java
package com.ieng.shortenLink.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.util.Date;

@Entity
@AllArgsConstructor
@RequiredArgsConstructor
@Getter
@Setter
@Table(name = "shorter_links")
public class ShorterLink {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "created_date")
    @CreationTimestamp
    private Date createdDate;

    @Column(name = "modified_date")
    @UpdateTimestamp
    private Date modifiedDate;

    @Column(name = "original_link", unique = true, nullable = false)
    private String originalLink;

    @Column(name = "hash", unique = true, nullable = false)
    private String hash;

    @Column(name = "expiry_date")
    private Date expiryDate;

}

File: shorten-link-ui/src/app/model/shorter-link.ts
export class ShorterLink {
    id!: number;
    createdDate!: Date;
    modifiedDate!: Date;
    expiryDate!: Date;
    hash!: string;
    originalLink!: string;

    constructor(originalLink: string, expiryDate: Date) {
        this.expiryDate = expiryDate;
        this.originalLink = originalLink;
    }
}
